tests/test_gf_unpack2.F90

```diff
@@ -0,0 +1,107 @@
+    & char(0), char(0), char(0), char(k10), char(0), char(0),  &
+    & char(0), char(30), char(1), char(0), char(k1), char(k1), &
+    & char(k1), char(k1), char(0), char(0), char(0), char(0),  &
+    & char(0), char(0), char(0), char(0), char(0), char(0),    &
+    & char(k2), char(k2), char(k2), char(k2), char(k3),         &
+    & char(k3), char(k3), char(k3), char(k4), char(k4),         &
+    & char(k4), char(k4),char(k5),char(k5), char(k5),           &
+    & char(k5), char(0), char(k6), char(k6), char(k6),          &
+    & char(k6), char(k7), char(k7), char(k7), char(k7),         &
+    & char(k8), char(k8), char(k8), char(k8), char(k8),         &
+    & char(k8), char(k8), char(k8), char(0), char(80),          &
+    & char(k6), char(k6), char(k6), char(k6), char(k6),         &
+    & char(k6), char(k6), char(k6), char(k9), char(k9),         &
+    & char(k9), char(k9), char(0), char(0), char(0), char(0),  &
+        ! section 4
+    & char(0), char(0), char(0), char(71), char(4), char(0),   &
+    & char(0), char(0), char(9), char(1), char(8), char(2),    &
+    & char(0), char(0), char(0), char(0), char(0), char(0),    &
+    & char(0), char(3), char(6), char(9), char(1), char(0),    &
+    & char(0), char(0), char(0), char(0), char(0), char(0),    &
+    & char(0), char(0), char(0),  char(0), char(0), char(0),   &
+    & char(1), char(0), char(0), char(0), char(0), char(0),    &
+    & char(3), char(0), char(0), char(0), char(n2), char(7),   & 
+    & char(n1), char(11), char(15), char(12), char(20),         &
+    & char(10), char(1), char(0), char(0), char(0), char(0),   &
+    & char(1), char(0), char(1), char(0), char(0), char(0),    &
+    & char(12), char(1), char(0), char(0), char(0), char(0),   &
+        ! section 5
+    & char(0), char(0), char(0), char(47), char(5),  char(0),  &
+    & char(0), char(0), char(k10), char(0), char(2),  char(0), &
+    & char(0), char(0), char(0), char(0), char(0),  char(0),   &
+    & char(1), char(8), char(1), char(1), char(1), char(Nan),  &
+    & char(Nan), char(Nan), char(Nan), char(0), char(0),        &
+    & char(0), char(0), char(0), char(0), char(0), char(0),    &
+    & char(0), char(0), char(0), char(0), char(0), char(0),    &
+    & char(0), char(0), char(0), char(0), char(0), char(0),    &
+        ! section 6
+    & char(0), char(0), char(0), char(6), char(6),  char(n3),  &
+        ! section 7
+    & char(0), char(0), char(0), char(12), char(7),  char(1),  &
+    & char(0), char(0), char(0), char(0), char(0),  char(0),   &
+        ! section 8
+    & "7", "7", "7", "7" /) 
+
+    iofst = 49
+    allocate(csec2(10))
+    print *, 'Testging gf_unpack2'
+    ! Normal grib message with offset to section 3
+    call gf_unpack2(fgrib, fgrib_len, iofst, lensec2, csec2, ierr)
+    if (ierr .ne. 6) stop 1
+    iofst = 38
+    call gf_unpack2(fgrib, fgrib_len, iofst, lensec2, csec2, ierr)
+    print *, 'ierr: ', ierr
+    deallocate(csec2)
```

Comment:
This is good to check the error handling code that was not being checked.

But can you also include a call to gf_unpack2 which works, and check the results are correct?

I know this is already done elsewhere, but in future, when wanting to look for a simple test, test_gf_unpack2.F90 is where we will look...

Reply:
Can you include an additional test, and then we can get this merged?

Reply:
???